openmc/data/fission_energy.py

```diff
@@ -463,11 +229,109 @@ def from_endf(cls, ev, incident_neutron):
+                # something is wrong.
+                second_order = coeffs[2]
+                if abs(second_order) * 1e12 > 1e8:
+                    # If we found the error, reduce 2nd-order coeff by 10**6.
+                    coeffs[2] /= EV_PER_MEV
+
+            # If multiple coefficients were given, we can create the polynomial
+            # and move on to the next component
+            if npoly > 0:
+                functions[name] = Polynomial(coeffs)
+                continue
+
+            # If a single coefficient was given, we need to use the Sher-Beck
+            # formula for energy dependence
+            zeroth_order = coeffs[0]
+            if name in ('delayed_photons', 'betas'):
+                func = Polynomial((zeroth_order, -0.075))
+            elif name == 'neutrinos':
+                func = Polynomial((zeroth_order, -0.105))
+            elif name == 'prompt_neutrons':
+                # Prompt neutrons require nu-data.  It is not clear from ENDF-102
+                # whether prompt or total nu value should be used, but the delayed
+                # neutron fraction is so small that the difference is negligible.
+                # MT=18 (n, fission) might not be available so try MT=19 (n, f) as
+                # well.
+                if 18 in incident_neutron.reactions:
+                    nu = [p.yield_ for p in incident_neutron[18].products
+                        if p.particle == 'neutron'
+                        and p.emission_mode in ('prompt', 'total')]
+                elif 19 in incident_neutron.reactions:
+                    nu = [p.yield_ for p in incident_neutron[19].products
+                        if p.particle == 'neutron'
+                        and p.emission_mode in ('prompt', 'total')]
+                else:
+                    raise ValueError('IncidentNeutron data has no fission '
+                                    'reaction.')
+                if len(nu) == 0:
+                    raise ValueError('Nu data is needed to compute fission energy '
+                                    'release with the Sher-Beck format.')
+                if len(nu) > 1:
+                    raise ValueError('Ambiguous prompt/total nu value.')
+
+                nu = nu[0]
+                if isinstance(nu, Tabulated1D):
+                    # Evaluate Sher-Beck polynomial form at each tabulated value
+                    func = deepcopy(nu)
+                    func.y = (zeroth_order + 1.307*nu.x - 8.07e6*(nu.y - nu.y[0]))
+                elif isinstance(nu, Polynomial):
+                    # Combine polynomials
+                    if len(nu) == 1:
+                        func = Polynomial([zeroth_order, 1.307])
+                    else:
+                        func = Polynomial(
+                            [zeroth_order, 1.307 - 8.07e6*nu.coef[1]]
+                            + [-8.07e6*c for c in nu.coef[2:]])
```

Comment:
Let's add another `elif` for fragments and set it to a linear polynomial with a slope of -0.25 as recommended by the latest ENDF-102.

Reply:
A little confused by this comment. In what I'm looking at, δEFR = 0.

Reply:
The latest ENDF-102 recommends assuming that ENF has a slope in the range -0.25 to -0.33 rather than the Sher-Beck assumption of 0.  However, I reread that section and it appears to really be advice for evaluators not ENDF users so I guess we should probably not mess with the Sher-Beck model.